src/datajudge/pytest_integration.py

```diff
@@ -4,9 +4,23 @@
 
 from .constraints.base import Constraint
 from .db_access import apply_patches
+from .formatter import AnsiColorFormatter, Formatter, HtmlFormatter
 from .requirements import Requirement
 
 
+@pytest.fixture(scope="session")
+def formatter(pytestconfig):
+    color = pytestconfig.getoption("color")
+    is_html = pytestconfig.getoption("htmlpath") is not None
+
+    if not is_html and (color == "yes" or color == "auto"):
+        return AnsiColorFormatter()
+    elif is_html and (color == "yes" or color == "auto"):
+        return HtmlFormatter()
+    else:
+        return Formatter()
```

Comment:
Could you add some tests that validate that this works?
e.g. something that creates a temp test file and calls `pytest` with different options on it.

Reply:
I have added an integration test to check if the formatter is selected correctly